delete-user-data/functions/src/index.ts

```diff
@@ -36,6 +39,56 @@ admin.initializeApp({
 
 logs.init();
 
+export const handleDeletion = functions.pubsub
+  .topic("deletions")
+  .onPublish(async (message, context) => {
+    const { path, uid } = JSON.parse(
+      Buffer.from(message.data, "base64").toString("utf8")
+    );
+
+    /** Delete collection if userId is included in the path */
+    if (path.includes(uid)) {
+      return deleteCollection(admin.firestore(), path, 100);
+    }
+
+    /** Check for subcollections and documents */
+    const collection = admin.firestore().collection(path);
+
+    /** Iterate through documents */
+    const docs = await collection.listDocuments();
+
+    docs.forEach(async (doc) => {
+      /** Check id value */
+      if (doc.path.includes(uid)) {
+        await doc.delete();
+      }
+
+      /** Check field/value */
+      const data = await doc.get();
+      const fields = data.data();
+      if (fields) {
+        Object.entries(fields).forEach(async ([field, value]) => {
+          if (field.includes(uid)) {
```

Comment:
Is this logic correct? You're deleting it if the document contains the Auth UID as a fieldname in the document?

I was thinking the user has to specify a auth_user_id_fieldname parameter (defaults to "userId"). The value of that is what should be matching Auth UID.

Reply:
Refactored with new query lookup.